src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,163 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level; _⊔_)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties          as ModuleProperties
+import Function.Definitions
+import Relation.Binary.Reasoning.Setoid   as Reasoning
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+  using (Σ; _,_; ∃; Σ-syntax; ∃-syntax; _×_)
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+-- module A = ModuleProperties A.leftSemimodule
+-- open ModuleProperties A.leftSemimodule
+--   using () renaming
+--   ( non-zeroʳ to non-zeroʳᴬ
+--   ; non-zeroˡ to non-zeroˡᴬ
+--   )
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+module B2 = ModuleProperties B.leftSemimodule
+-- open ModuleProperties B.leftSemimodule
+--   using () renaming
+--   ( non-zeroʳ to non-zeroʳᴮ
+--   ; non-zeroˡ to non-zeroˡᴮ
+--   )
+open CommutativeRing ring
+  using (_≈_; _*_; _+_; 0#) renaming (Carrier to S)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  -- f(x) ≈ 0 iff x ≈ 0, for linear non-trivial f
+  f𝟘≈𝟘 : f A.0ᴹ B.≈ᴹ B.0ᴹ
```

Comment:
Please could we use standard `0`s in the names rather than boldfaced ones. More standard, and easier to type!

Reply:
Changed as requested in next push.